### ISSUE TITLE
"Null" Class & Approximate Tokens

### ISSUE THREAD

**Issue by AlexIzydorczyk** [facebookresearch/fastText#60]
Hi,

2 quick questions that perhaps run the risk of  being more general than just specific fastText - but relating to the classification capability:

1) Given a corpus of short messages (something like Tweets), how (could?) one approach a problem where the vast majority of documents are of a "null class" while a minority are of a specific labelled class. It's possible to include in the training set of "null class" examples, however, the null class has a much wider array of possible tokens (including many unobserved tokens) 

Is there some way to get the pseudo-probability (rather than k most likely classes) from the predict function? Then if a given document did not reach some threshold, it could be considered as not strongly matching any labelled class. 

2) Does fastText support any concept rather than exact tokens? The use case here is spelling error (ie. we may have observed "AMAZON" but not "AMZON" despite them having presumably the same meaning). 

Thanks,
Alex

**Comment by rafael-aero** [comment]
Regarding 2), from what I tested here, it seems that fastText requires exact matches on words that existed on the training data-set. Any word it doesn't find in the training dictionary is just ignored when computing the predictions.

**Comment by AlexIzydorczyk** [comment]
I suppose that one reasonable way around this would be to replace out of vocabulary words with the closest in vocabulary word based on vector representation of both

**Comment by EdouardGrave** [comment]
Regarding 1), I believe that including "null class" examples in the training set is the best solution. We will also add an option to output the probability of predicted labels (probably today).

Regarding 2), another possibility is to used character n-gram features (as described in the paper "Enriching word vectors with subword information"). We will also add this functionality to the `supervised` setting soon.

**Comment by AlexIzydorczyk** [comment]
Great, looking forward to both features.
